pkg/controller/machinedeployment/rolling.go

```diff
@@ -34,6 +34,9 @@ func (r *ReconcileMachineDeployment) rolloutRolling(d *v1alpha1.MachineDeploymen
 	if err != nil {
 		return err
 	}
+	if newMS == nil {
```

Comment:
Can we add a comment here explaining when `newMS` is supposed to be nil and why?

Reply:
Does this imply that changes to annotations / minreadyseconds are applied retroactively to the old machine set?

Reply:
Yes.
See https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machinedeployment/sync.go#L109

Callstack:
```
(dlv) stack
0  0x00000000013e6930 in sigs.k8s.io/cluster-api/pkg/controller/machinedeployment.(*ReconcileMachineDeployment).getNewMachineSet
   at /go/src/sigs.k8s.io/cluster-api/pkg/controller/machinedeployment/sync.go:109
1  0x00000000013e6667 in sigs.k8s.io/cluster-api/pkg/controller/machinedeployment.(*ReconcileMachineDeployment).getAllMachineSetsAndSyncRevision
   at /go/src/sigs.k8s.io/cluster-api/pkg/controller/machinedeployment/sync.go:75
2  0x00000000013e3752 in sigs.k8s.io/cluster-api/pkg/controller/machinedeployment.(*ReconcileMachineDeployment).rolloutRolling
   at /go/src/sigs.k8s.io/cluster-api/pkg/controller/machinedeployment/rolling.go:33
3  0x00000000013e1bc2 in sigs.k8s.io/cluster-api/pkg/controller/machinedeployment.(*ReconcileMachineDeployment).Reconcile
   at /go/src/sigs.k8s.io/cluster-api/pkg/controller/machinedeployment/controller.go:182
```